crates/bevy_ecs/src/reflect/component.rs

```diff
@@ -83,14 +86,12 @@ pub struct ReflectComponent(ReflectComponentFns);
 /// world.
 #[derive(Clone)]
 pub struct ReflectComponentFns {
-    /// Function pointer implementing [`ReflectComponent::from_world()`].
-    pub from_world: fn(&mut World) -> Box<dyn Reflect>,
     /// Function pointer implementing [`ReflectComponent::insert()`].
-    pub insert: fn(&mut EntityWorldMut, &dyn Reflect),
+    pub insert: fn(&mut EntityWorldMut, &dyn Reflect, &TypeRegistry),
```

Comment:
I have added a `&TypeRegistry` parameter to `ReflectComponent`'s functions, mimicking the ones on `ReflectBundle`. Now the `TypeRegistry` in the `World` is no longer implicitly accessed.